File: src/controllers/grading.controller.ts
import { Response, NextFunction } from "express";
import { AuthenticatedRequest } from "../types";
import { GradingService } from "../services/grading.service";

export class GradingController {
  static async setGradingRules(req: AuthenticatedRequest, res: Response, next: NextFunction) {
    try {
      const schoolId = req.schoolId!;
      const result = await GradingService.setGradingRules(schoolId, req.body);
      if (result.success) {
        return res.status(200).json(result);
      }
      return res.status(400).json(result);
    } catch (error) {
      next(error);
    }
  }

  static async getGradingRules(req: AuthenticatedRequest, res: Response, next: NextFunction) {
    try {
      const schoolId = req.schoolId!;
      const result = await GradingService.getGradingRules(schoolId);
      return res.status(200).json(result);
    } catch (error) {
      next(error);
    }
  }
}

File: _old_csharp_backup/LeonEdBackend/Controllers/GradingController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using LeonEdBackend.DTOs.GradingDtos;
using LeonEdBackend.Services.Interfaces;

namespace LeonEdBackend.Controllers
{
    [Route("api/[controller]")]
    [Authorize]
    public class GradingController : BaseController
    {
        private readonly IGradingService _gradingService;

        public GradingController(IGradingService gradingService)
        {
            _gradingService = gradingService;
        }

        /// <summary>Set grading rules for the school (replaces existing rules)</summary>
        [HttpPost("rules")]
        [Authorize(Roles = "SchoolAdmin")]
        public async Task<IActionResult> SetGradingRules([FromBody] BulkCreateGradingRulesRequest request)
        {
            var schoolId = RequireSchoolId();
            var result = await _gradingService.SetGradingRules(schoolId, request);
            return result.Success ? Ok(result) : BadRequest(result);
        }

        /// <summary>Get grading rules (returns defaults if not configured)</summary>
        [HttpGet("rules")]
        [Authorize(Roles = "SchoolAdmin,Teacher")]
        public async Task<IActionResult> GetGradingRules()
        {
            var schoolId = RequireSchoolId();
            var result = await _gradingService.GetGradingRules(schoolId);
            return Ok(result);
        }
    }
}
